### ISSUE TITLE
ui: replication status in DB Console Overview fails to report missing ranges 

### ISSUE THREAD

**Issue by dhartunian** [cockroachdb/cockroach#91165]
**Describe the problem**

When a table's primary region goes down, the ranges for that table do **not** get reported as unavailable in the DB Console overview page.

**To Reproduce**

via @chriscasano 
```
CLUSTER="chrisc-ml"
roachprod create ${CLUSTER} --nodes 9 -c aws --aws-zones=us-east-1a,us-east-2a,ca-central-1a
roachprod stage ${CLUSTER} workload
roachprod stage ${CLUSTER} release latest

roachprod sql ${CLUSTER}:1
set cluster setting server.time_until_store_dead='1m20s';
create database test;
use test;
show regions from cluster;
alter database test primary region "us-east-1";
alter database test add region "us-east-2";
alter database test add region "ca-central-1";
create table t (i int primary key default unique_rowid(), s string) placement restricted;

ALTER TABLE users SPLIT AT VALUES (100000000000000000), (200000000000000000), (300000000000000000), (400000000000000000), (500000000000000000), (600000000000000000), (700000000000000000), (800000000000000000), (900000000000000000);

insert into table t values ('a'),('b'),('c'),('d'),('e'),('f'),('g'),('h'),('i'),('j');

show ranges from table t;

<screenshot UI>

roachprod stop ${CLUSTER}:1-3

<screenshot UI>

# Wait 1.5 minutes

<screenshot UI>
```

**Expected behavior**

We would expect a non-zero number of unavailable ranges in the DB Console

**Additional data / screenshots**

**Environment:**
 - CockroachDB version: any
 - Server OS: Linux
 - Client app: DB Console

Jira issue: CRDB-21131

Epic CRDB-28775

**Comment by ajwerner** [comment]
Which UI page are you thinking of? Is this related to https://github.com/cockroachdb/cockroach/pull/91130?

**Comment by dhartunian** [comment]
@ajwerner the replication status box on the landing page. seems unrelated to the ticket you linked.

<img width="410" alt="Screen Shot 2022-11-03 at 9 22 38 AM" src="https://user-images.githubusercontent.com/986307/199731981-d80885d9-6c6c-4c47-b753-e775bb3cebd0.png">

**Comment by chriscasano** [comment]
Let me rephrase this a bit.  The issue we're seeing is that in a regional failure, we see unavailable ranges, but after the nodes become dead, the unavailable ranges in the Console UI go down to zero.  We say this using the new SRE capability for simulating a regional failure by isolating the network of the pods, I believe this was done using a network policy.   However, when I tried the same failure steps using self-hosted (the steps above), I didn't see the same behavior.  Here are the screen shots from the SRE test:

Shows zone config the ranges that are unavailable in East 1.  The failure test failed East 1

![IMG_4255](https://user-images.githubusercontent.com/1087114/199739858-857fa37c-9d4d-4d98-80fa-29b7db8a9213.jpg)

Shows the ranges for the user table that will become unavailable:


![IMG_4256](https://user-images.githubusercontent.com/1087114/199739356-a8dc37b0-4ac1-4fcc-82f5-0483198380fe.jpg)

Console UI showing the 79 unavailable ranges when the nodes go suspect

![IMG_4257](https://user-images.githubusercontent.com/1087114/199739358-22198c71-adcd-4f7f-bf16-6dcee0471050.jpg)

Console UI showing 0 unavailable ranges when the nodes go dead.

![IMG_4258](https://user-images.githubusercontent.com/1087114/199739360-37eccd9c-9d8e-4dea-a8a6-a98f8d973b8c.jpg)

When I tried this out with the roachprod steps above using self hosted, I couldn't recreate the issue above.  However, I didn't use placement restricted to keep all of the nodes in East 1 which may be another good test.

**Comment by chriscasano** [comment]
@lunevalex FWIW - I tried the same test above but using placement restricted for East 1, and then brought down all nodes in East 1.  The status metrics got really weird.  I see under replicated ranges...which seems weird.  And nothing shows for Unavailable Ranges.

```
SET enable_multiregion_placement_policy = on;
SET CLUSTER SETTING sql.defaults.multiregion_placement_policy.enabled = on;
ALTER DATABASE test PLACEMENT RESTRICTED;
```

root@localhost:26257/test> select replicas from [show ranges from table t];
  replicas
------------
  {1,2,3}
  {1,2,3}
  {1,2,3}
  {1,2,3}
  {1,2,3}
  {1,2,3}
  {1,2,3}
  {1,2,3}
  {1,2,3}
  {1,2,3}
(10 rows)

Once nodes are dead the under-replicated metrics slowly go to zero.

![Screen Shot 2022-11-04 at 4 33 46 PM](https://user-images.githubusercontent.com/1087114/200071254-de32c823-f9d1-400f-b917-527e8010eb05.png)

![Screen Shot 2022-11-04 at 4 36 02 PM](https://user-images.githubusercontent.com/1087114/200071290-7fbf7911-0050-4807-a814-b55856448f6e.png)

I would say the good news is that Andrei is right.  The replication reports seem accurate:


```
root@localhost:26257/defaultdb> select zone_id, total_ranges, unavailable_ranges from system.replication_stats;
  zone_id | total_ranges | unavailable_ranges
----------+--------------+---------------------
        0 |            1 |                  0
        1 |           39 |                  0
       16 |            1 |                  0
       17 |            2 |                  0
       22 |            1 |                  0
      **104 |           10 |                 10**
(6 rows)
```

So I guess the UI UX experience could probably use some love.
